src/view/configuration/configuration.jsx

```diff
@@ -143,6 +136,12 @@ const getSettings = async ({ values, initInfo }) => {
       }),
     ),
   };
+
+  if (values.instances.some((instance) => instance.useExistingAlloy)) {
+    settings.hostedLibFiles = [];
+  }
+
+  return settings;
```

Comment:
Near as I can tell, you are only ever setting this to []. If you want to use hostedLibFiles, you have to change the build and package process to generate another file. Seems we'll need to change this for every build though. We need to have a discussion if this will affect page performance for people not including their own Web SDK.

Reply:
My understanding, and I might be wrong here, is that setting it here basically overrides the backend logic that would otherwise embed `alloy.js` by default. So by forcibly emptying the array here when the toggle is set, I end up removing `alloy.js` from the build (since it's already self-hosted).

Reply:
hostedLibFiles is a property at the root level of the extension manifest, but you are changing the extension configuration settings here. Additionally, when you build the extension, the Web SDK files are packaged inside the library bundle, it is not a hosted lib file.